kie-server-parent/kie-server-services/kie-server-services-task-assigning/kie-server-services-task-assigning-planning/src/main/java/org/kie/server/services/taskassigning/planning/SolutionSynchronizer.java

```diff
@@ -119,12 +148,32 @@ public void initSolverExecutor() {
         startPermit.release();
     }
 
+    /**
+     * Starts the synchronization of the solution from the indicated last modification date.
+     * @param solution a non null solution instance to synchronize.
+     * @param fromLastModificationDate filtering parameter for reading the modifications.
+     */
     public void synchronizeSolution(TaskAssigningSolution solution, LocalDateTime fromLastModificationDate) {
+        synchronizeSolution(solution, fromLastModificationDate, Duration.ofMillis(0));
+    }
+
+    /**
+     * Starts the synchronization of the solution from the indicated last modification date.
+     * @param solution a non null solution instance to synchronize.
+     * @param fromLastModificationDate filtering parameter for reading the modifications.
+     * @param unchangedPeriodTimeout a non null period of time for returning from the synchronization if no changes were
+     * produced during that period. A negative or zero period is ignored.
```

Comment:
same as above, nobody will configure things like "wait -10 seconds". I think it's ok.

Reply:
Most often no, but when somebody does, it's better to give him a nudge instead of silently ignoring wrong input.

Accepting unsanitized inputs leads to unnecessary complexity of the code doing calculations based on it, like here:
https://github.com/kiegroup/droolsjbpm-integration/pull/2181/files/8744ee8013894c78258d26726e3da7de98aaeb95#diff-286401b0213653b16fad9e16daab206cR341-R345

Reply:
was applied